extensions/himalaya/src/index.tsx

```diff
@@ -0,0 +1,414 @@
+  };
+  subject: string;
+  date: Date;
+}
+
+interface Folder {
+  delim: string;
+  name: string;
+  desc: string;
+}
+
+interface State {
+  isLoading: boolean;
+  envelopes: Envelope[];
+  folders: Folder[];
+}
+export default function ListEnvelopes() {
+  const [state, setState] = useState({
+    isLoading: true,
+    envelopes: [],
+    folders: [],
+  } as State);
+
+  useEffect(() => {
+    async function fetch() {
+      const envelopes = await listEnvelopes();
+      const folders = await listFolders();
+
+      setState((previous: State) => ({
+        ...previous,
+        envelopes: envelopes,
+        folders: folders,
+        isLoading: false,
+      }));
+    }
+
+    fetch();
+  }, []);
+
+  const accessories = (envelope: Envelope) => {
+    const accessories = [];
+
+    if (envelope.flags.includes(Flag.Seen)) {
+      accessories.push({
+        icon: Icon.Circle,
+      });
+    }
+
+    if (!envelope.flags.includes(Flag.Seen)) {
+      accessories.push({
+        icon: Icon.CircleFilled,
+      });
+    }
+
+    if (envelope.flags.includes(Flag.Answered)) {
+      accessories.push({
+        icon: Icon.Reply,
+      });
+    }
+
+    // This is always present. Accessories are rendered ordered
+    // so put them last.
+    accessories.push({
+      text: {
+        value: `${envelope.from.name} <${envelope.from.addr}>`,
+      },
+      icon: Icon.Person,
+    });
+
+    return accessories;
+  };
+
+  const markUnreadAction = (envelope: Envelope) => {
+    return (
+      <Action
+        title="Mark Unread"
+        style={Action.Style.Regular}
+        icon={Icon.CircleFilled}
+        onAction={async () => {
+          const toast = await showToast({
+            style: Toast.Style.Animated,
+            title: "Marking as unread",
+          });
+
+          try {
+            const { stdout, _stderr } = await exec(`"himalaya" flag remove ${envelope.id} -- seen`, {
+              env: {
+                PATH: PATH,
+              },
+            });
+
+            toast.style = Toast.Style.Success;
+            toast.title = "Marked unread";
+
+            setState((previous: State) => ({ ...previous, isLoading: true }));
+            const envelopes = await listEnvelopes();
+            setState((previous: State) => ({ ...previous, envelopes: envelopes, isLoading: false }));
+          } catch (error) {
+            toast.style = Toast.Style.Failure;
+            toast.title = `Failed to mark unread: ${error}`;
+          }
+        }}
+      />
+    );
+  };
+
+  const markReadAction = (envelope: Envelope) => {
+    return (
+      <Action
+        title="Mark Read"
+        style={Action.Style.Regular}
+        icon={Icon.Circle}
+        onAction={async () => {
+          const toast = await showToast({
+            style: Toast.Style.Animated,
+            title: "Marking as read",
+          });
+
+          try {
+            const { stdout, _stderr } = await exec(`"himalaya" flag add ${envelope.id} -- seen`, {
```

Comment:
It could also be great if Himalaya could export Node bindings (I found https://github.com/neon-bindings/neon, looks awesome).

Reply:
That does look interesting! I how they would be used with Himalaya CLI?

Reply:
No idea yet, sth to think well! Not sure if it would be simpler than just using the CLI directly :thinking:

Reply:
Haha fair point. You're right though, it's still something to consider!